e2e/src/auction_bid.rs

```diff
@@ -0,0 +1,87 @@
+use ethcontract::web3::types::{H160, H256, U256};
+
+#[derive(Debug)]
+pub struct AuctionBid {
```

Comment:
You could just have a

`pub struct AuctionBid(([u8; 32], ...))`

and then:

```rs
impl AuctionBid {
  pub fn order_hash(&self) -> H256 {
    H256::from_slice(self.0)
  }
  ...
```

That should give you a lightweight named view on top of this monster tuple.

Reply:
Yes, I like this, however, clippy will insist that I declare a type for the tuple consisting of `AuctionBidData`. Would love if it were possible to declare functions for types instead of structs. Will check and see if this is possible.